app/assets/javascripts/controllers/middleware_topology/middleware_topology_controller.js

```diff
@@ -112,8 +150,25 @@ function MiddlewareTopologyCtrl($scope, $http, $interval, $location, topologySer
             default :
                 return { x: -9, y: -9, height: 18, width: 18, r: 17};
         }
+    }
 
-    };
+    self.getTextPosition = function getTextPosition(d) {
```

Comment:
Hmm, in fact it's used only for positioning the glyphs, it's not related to the labels (but I'll double check). I'll move that to the css and remove this function. Good catch